src/Utils.ts

```diff
@@ -12,75 +14,102 @@ if (window.ethereum) {
-        const balance = await rifTokenContract.methods.balanceOf(address).call();
+        const balance = await rifTokenContract.methods
+            .balanceOf(address)
+            .call();
         return balance;
     }
-    static async getTokenContract( ) {
-        let rifTokenContract = new web3.eth.Contract(TestToken.abi, process.env.REACT_APP_CONTRACTS_RIF_TOKEN);
+
+    static async getTokenContract() {
+        const rifTokenContract = new web3.eth.Contract(
+            TestToken.abi,
+            process.env.REACT_APP_CONTRACTS_RIF_TOKEN
+        );
         rifTokenContract.setProvider(web3.currentProvider);
         return rifTokenContract;
     }
 
-    static async getBalance(address) {
+    static async getBalance(address: string) {
         const balance = await web3.eth.getBalance(address);
         return balance;
     }
-    static fromWei(balance) {
+
+    static fromWei(balance: string) {
         return web3.utils.fromWei(balance);
     }
 
-    static async getReceipt(transactionHash) {
-        let receipt = await web3.eth.getTransactionReceipt(transactionHash)
-        let times = 0
+    static async getReceipt(transactionHash: string) {
+        let receipt = await web3.eth.getTransactionReceipt(transactionHash);
+        let times = 0;
 
         while (receipt === null && times < 40) {
-            times++
-            const sleep = new Promise(resolve => setTimeout(resolve, 30000))
-            await sleep
-            receipt = await web3.eth.getTransactionReceipt(transactionHash)
+            times += 1;
+            // eslint-disable-next-line no-promise-executor-return
+            const sleep = new Promise((resolve) => setTimeout(resolve, 30000));
+            // eslint-disable-next-line no-await-in-loop
+            await sleep;
+            // eslint-disable-next-line no-await-in-loop
+            receipt = await web3.eth.getTransactionReceipt(transactionHash);
         }
 
-        return receipt
+        return receipt;
     }
 
-    static async getAccounts() {
-        const accounts = await web3.eth.getAccounts()
+    static async getAccounts(): Promise<string[]> {
+        const accounts = await web3.eth.getAccounts();
         if (accounts.length === 0) {
-            console.error("Couldn't get any accounts! Make sure your Client is configured correctly.")
-            return
+            console.error(
+                "Couldn't get any accounts! Make sure your Client is configured correctly."
+            );
+            return [];
         }
         return accounts;
     }
 
-    static async toWei(tRifPriceInRBTC) {
+    static async toWei(tRifPriceInRBTC: string) {
         return web3.utils.toWei(tRifPriceInRBTC);
     }
 
-    static async getTransactionReceipt(transactionHash) {
-        return await web3.eth.getTransactionReceipt(transactionHash);
+    static async getTransactionReceipt(transactionHash: string) {
+        return web3.eth.getTransactionReceipt(transactionHash);
```

Comment:
yes, unless we wanted to handle errors within this function (and that would require a try/catch). The user who will call the `getTransactionReceipt` is going to use the await (since it's an async function), so we are already returning a promise that will be awaited directly from the final user.
```ts
const receipt = await getTransactionReceipt() // it returns a promise regardless of the internal await
```
So we let the user wait for the real promise instead of letting the user wait for a promise that resolves just after the internal one.
Not sure if it's clear 😅

Reply:
clear as day :+1: